File: frontend/src/app/core/services/admin-staff.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpParams } from '@angular/common/http';
import { ApiService } from './api.service';
import { pageParams } from '../../shared/utils/page-params.util';
import {
  StaffMember,
  StaffKpis,
  StaffRolesMatrix,
  StaffCandidate,
  StaffFilters,
  Page,
} from '../../models/admin-staff.model';

/**
 * Platform-staff API client for the Staff & Roles console: the staff roster (filterable + paginated),
 * role KPIs, the read-only role→capability matrix, grant-access candidates, and grant / change-role /
 * revoke mutations. All gate on STAFF_MANAGE (Super Admin only); self-change is rejected + audit-logged.
 */
@Injectable({ providedIn: 'root' })
export class AdminStaffService {
  private readonly api = inject(ApiService);
  private readonly base = '/admin/staff';

  roster(filters: StaffFilters = {}, page = 0, size = 10) {
    const params = pageParams(filters, page, size);
    return this.api.get<Page<StaffMember>>(this.base, { params });
  }

  kpis() { return this.api.get<StaffKpis>(`${this.base}/kpis`); }
  roles() { return this.api.get<StaffRolesMatrix>(`${this.base}/roles`); }
  candidates(search: string) {
    const params = new HttpParams().set('search', search);
    return this.api.get<StaffCandidate[]>(`${this.base}/candidates`, { params });
  }

  /** Grant or change a staff member's role. */
  setRole(userId: string, role: string) { return this.api.post<StaffMember>(`${this.base}/${userId}`, { role }); }
  /** Revoke all staff access. */
  revoke(userId: string) { return this.api.delete<StaffMember>(`${this.base}/${userId}`); }
}

File: backend/src/main/java/com/postwerk/service/AdminStaffService.java
package com.postwerk.service;

import com.postwerk.dto.admin.StaffCandidateResponse;
import com.postwerk.dto.admin.StaffKpisResponse;
import com.postwerk.dto.admin.StaffMemberResponse;
import com.postwerk.dto.admin.StaffRolesMatrixResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

/**
 * Platform-staff Staff &amp; Roles console: the staff roster, role KPIs, the read-only role→capability
 * matrix, grant-access candidates, and grant / change-role / revoke mutations. View + mutate gate on
 * {@code STAFF_MANAGE} (Super Admin only); self-change is rejected; every mutation is audit-logged.
 *
 * @since 1.0
 */
public interface AdminStaffService {

    Page<StaffMemberResponse> listRoster(String search, String role, String tier, String sort,
                                         String actorEmail, Pageable pageable);

    StaffKpisResponse kpis();

    StaffRolesMatrixResponse rolesMatrix();

    List<StaffCandidateResponse> candidates(String search);

    /** Grant (old role null) or change a staff role. */
    StaffMemberResponse setRole(UUID userId, String role, String actorEmail, String ip);

    /** Revoke all staff access (clears the role). */
    StaffMemberResponse revoke(UUID userId, String actorEmail, String ip);
}
